packages/shared/__tests__/define-vars/stylex-define-vars-test.js

```diff
@@ -261,6 +261,82 @@ describe('stylex-define-vars test', () => {
     `);
   });
 
+  test('converts set of vars with nested at rules to CSS and does not include key prefix in debug mode with debug classnames off', () => {
+    const themeName = 'TestTheme.stylex.js//buttonTheme';
+    const classNamePrefix = 'x';
+    const defaultVars = {
+      bgColor: {
+        default: 'blue',
+        '@media (prefers-color-scheme: dark)': {
+          default: 'lightblue',
+          '@supports (color: oklab(0 0 0))': 'oklab(0.7 -0.3 -0.4)',
+        },
+        '@media print': 'white',
+      },
+      bgColorDisabled: {
+        default: {
+          default: 'grey',
+          '@supports (color: oklab(0 0 0))': 'oklab(0.7 -0.3 -0.4)',
+        },
+        '@media (prefers-color-scheme: dark)': {
+          default: 'rgba(0, 0, 0, 0.8)',
+          '@supports (color: oklab(0 0 0))': 'oklab(0.7 -0.3 -0.4)',
+        },
+      },
+      cornerRadius: '10px',
+      fgColor: {
+        default: 'pink',
+      },
+    };
+    const [jsOutput, cssOutput] = styleXDefineVars(defaultVars, {
+      themeName,
+      enableDebugClassNames: false,
+      debug: false,
+    });
+
+    expect(jsOutput).toEqual({
+      __themeName__: classNamePrefix + createHash(themeName),
+      bgColor: `var(--${classNamePrefix + createHash(`${themeName}.bgColor`)})`,
+      bgColorDisabled: `var(--${
+        classNamePrefix + createHash(`${themeName}.bgColorDisabled`)
+      })`,
+      cornerRadius: `var(--${
+        classNamePrefix + createHash(`${themeName}.cornerRadius`)
+      })`,
+      fgColor: `var(--${classNamePrefix + createHash(`${themeName}.fgColor`)})`,
+    });
```

Comment:
Do you think we could use an inline snapshot here instead of reproducing implementation details?

Reply:
This test was written this way to verify that the _correct_ hash was generated.
In the past there was a bug where two different parts of the codebase ended up generating distinct hashes.